src/utils/lessons-utils.js

```diff
@@ -0,0 +1,43 @@
+import moment from 'moment';
+
+const determineUpcomingLesson = (now, lessons) => {
+  const timeOfChecking = now instanceof Date ? now.toISOString() : now;
+
+  const upcomingLesson = lessons.find((lesson, index) => {
+    const startsOn = lesson.schedule?.startsOn;
+
+    const isFutureLesson = startsOn >= timeOfChecking;
+    const previousLessonStartsOn = lessons[index - 1]?.schedule?.startsOn;
+    const previousLessonWasPastLesson = !previousLessonStartsOn || previousLessonStartsOn < timeOfChecking;
+    return isFutureLesson && previousLessonWasPastLesson;
+  });
+
+  return upcomingLesson;
+};
+
+const translateTimeUnit = (timeUnit, t, translationKey) => {
```

Comment:
I wonder 2 things:

* could the creation of the string "in 2 days" etc. be done by https://momentjs.com/docs/#/displaying/tonow/?
* should this helper function rather go into `time-utils.js` (and/or into the `language-context.js` where there's already `formatDate`?

Reply:
It works ("time to X" where X is our future date, because "time to now" treats the date given as being in the past). I somehow wanted to go for the **exact time until** (e.g. "2 days, 3 hours, 5 minutes") but that's actually not necessary.

Thanks for suggesting this, simplifies the code alot.